account/client_test.go

```diff
@@ -3,202 +3,81 @@ package account
-			DOB: stripe.DOB{
-				Day:   1,
-				Month: 2,
-				Year:  1990,
-			},
-		},
-		TOSAcceptance: &stripe.TOSAcceptanceParams{
-			IP:        "127.0.0.1",
-			Date:      1437578361,
-			UserAgent: "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_4) AppleWebKit/600.7.12 (KHTML, like Gecko) Version/8.0.7 Safari/600.7.12",
-		},
-	}
-
-	_, err := New(params)
-	if err != nil {
-		t.Error(err)
-	}
+func TestAccountGet(t *testing.T) {
+	account, err := Get()
+	assert.Nil(t, err)
+	assert.NotNil(t, account)
 }
 
-func TestAccountLegalEntity(t *testing.T) {
-	params := &stripe.AccountParams{
-		Type:    stripe.AccountTypeCustom,
-		Country: "US",
-		LegalEntity: &stripe.LegalEntity{
-			Type:          stripe.Company,
-			BusinessTaxID: "111111",
-			SSN:           "1111",
-			PersonalID:    "111111111",
-			DOB: stripe.DOB{
-				Day:   1,
-				Month: 2,
-				Year:  1990,
-			},
-		},
-	}
-
-	target, err := New(params)
-	if err != nil {
-		t.Error(err)
-	}
-
-	if !target.LegalEntity.BusinessTaxIDProvided {
-		t.Errorf("Account is missing BusinessTaxIDProvided even though we submitted the value.\n")
-	}
-
-	if !target.LegalEntity.SSNProvided {
-		t.Errorf("Account is missing SSNProvided even though we submitted the value.\n")
-	}
-
-	if !target.LegalEntity.PersonalIDProvided {
-		t.Errorf("Account is missing PersonalIDProvided even though we submitted the value.\n")
-	}
+func TestAccountGetByID(t *testing.T) {
+	account, err := GetByID("acct_123", nil)
+	assert.Nil(t, err)
+	assert.NotNil(t, account)
 }
 
-func TestAccountDelete(t *testing.T) {
-	params := &stripe.AccountParams{
-		Type:                 stripe.AccountTypeCustom,
-		Country:              "CA",
-		BusinessUrl:          "www.stripe.com",
-		BusinessName:         "Stripe",
-		BusinessPrimaryColor: "#ffffff",
-		SupportEmail:         "foo@bar.com",
-		SupportUrl:           "www.stripe.com",
-		SupportPhone:         "4151234567",
-		LegalEntity: &stripe.LegalEntity{
-			Type:         stripe.Individual,
-			BusinessName: "Stripe Go",
-			DOB: stripe.DOB{
-				Day:   1,
-				Month: 2,
-				Year:  1990,
-			},
-		},
-	}
-
-	acct, err := New(params)
-	if err != nil {
-		t.Error(err)
-	}
-
-	acctDel, err := Del(acct.ID, nil)
-	if err != nil {
-		t.Error(err)
-	}
+func TestAccountList(t *testing.T) {
+	i := List(&stripe.AccountListParams{})
 
-	if !acctDel.Deleted {
-		t.Errorf("Account id %q expected to be marked as deleted on the returned resource\n", acctDel.ID)
-	}
+	// Verify that we can get at least one account
+	assert.True(t, i.Next())
+	assert.Nil(t, i.Err())
+	assert.NotNil(t, i.Account())
 }
 
-func TestAccountReject(t *testing.T) {
-	params := &stripe.AccountParams{
+func TestAccountNew(t *testing.T) {
+	account, err := New(&stripe.AccountParams{
 		Type:                 stripe.AccountTypeCustom,
 		Country:              "CA",
 		BusinessUrl:          "www.stripe.com",
 		BusinessName:         "Stripe",
 		BusinessPrimaryColor: "#ffffff",
+		DebitNegativeBal:     true,
```

Comment:
How does default work with the new system? I don't think anything has changed but wanted to ask since I see you add it explicitly.

Reply:
So it's essentially identical to the old way. For a boolean like this one we only encode if the value is `true`, so like before, we still need the inverted "No" field (e.g. `NoDebitNegativeBal`) to encode a false.